In [2]:
from IPython.display import IFrame
import math
import numpy as np

# Coordinate systems

Motion (change of position in space over time) is not an absolute concept; we need a reference point relative to which we describe the movement of an object. Likewise, the state of this reference cannot be absolute in space, and therefore motion is relative.

A reference frame is a place relative to which we choose to describe the motion of an object. In this reference frame, we define a coordinate system (set of axes) in which we measure the motion of an object (but the terms reference frame and coordinate system are often used interchangeably).

The choice of a reference frame and a coordinate system is often a matter of judgment and practicality. However, there is an important distinction between reference frames when we are dealing with the dynamics of motion, where we are interested in the forces associated with the motion of an object. In dynamics, we speak of an inertial frame (also known as a Galilean frame) when Newton's laws of motion in their simplest form apply in that frame, and a non-inertial frame (also known as a non-inertial reference frame) when Newton's laws in their simplest form do not apply (apparent accelerations/forces appear in such a frame). An inertial frame is at rest or moving at a constant velocity (since there is no such thing as absolute rest!), while a non-inertial frame is accelerated (relative to an inertial frame).

The concept of a frame of reference has changed significantly since the days of Aristotle, Galileo, Newton, and Einstein. You can read more about this and the philosophical implications in the article [Space and Time: Inertial Frames]( http://plato.stanford.edu/entries/spacetime-iframes/).

## Coordinate system for human movement analysis

In anatomy, we use a simplified reference system composed of perpendicular planes to provide a standard reference for qualitatively describing the structures and movements of the human body, as shown in the following figure.

<div class='center-align'><figure><img src="http://upload.wikimedia.org/wikipedia/commons/3/34/BodyPlanes.jpg" width="300" alt="Anatomical body position"/><figcaption><center><i>Figure. Anatomical body position and body planes (<a href="http://en.wikipedia.org/wiki/Human_anatomical_terms" target="_blank">image from Wikipedia</a> ).< /i></center></figcaption> </figure></div>

## Coordinate system base

**The base of a coordinate system** is a set of vectors that allow us to uniquely describe the position of any point in space. Think of it as a set of axes that intersect at a single point (the origin) and determine the directions in which we measure distances.

**Why do we need a base?**

* **Unambiguous description of position:** Thanks to the base, we can uniquely identify each point in space using coordinates that indicate how far this point is from the origin in the direction of each axis.
* **Simplification of calculations:** The base allows us to convert geometric problems into algebraic ones, which makes them much easier to solve.
* **Standardization:** The base provides us with a common language for describing spatial relationships.

**What does the base look like?**

* **In a plane:** The base is formed by two linearly independent vectors that determine the direction of the x and y axes. The most commonly used is the Cartesian coordinate system, where the x and y axes are perpendicular to each other and have the same unit.
   
* **In space:** The base consists of three linearly independent vectors that determine the direction of the x, y, and z axes. Again, the Cartesian coordinate system is most often used, where the axes are mutually perpendicular and have the same unit.
   

**Base features:**
* **Linear independence:** No base vector must be a linear combination of the other base vectors.
* **Range:** The number of vectors in the base is equal to the dimension of the space. In the plane there are two vectors, in space there are three.

## Orthogonal and orthonormal bases

An **orthogonal base** is a special type of base for a vector space where all vectors are perpendicular to each other. This means that the dot product of any two distinct vectors in that base is equal to zero.

**Why is an orthogonal base advantageous?**

* **Simpler calculations:** Many calculations, such as projecting vectors or finding the coordinates of a vector relative to a base, are significantly simpler for orthogonal bases.
* **Geometric interpretation:** Orthogonality of vectors has a direct geometric interpretation - the vectors form a right angle.

An **orthonormal base** is an even more special case of an orthogonal base. In addition to all vectors in this base being mutually perpendicular, they also have unit length (i.e. their magnitude is equal to one).

**Why is an orthonormal base advantageous?**

* **Normalization:** Unit length of vectors simplifies many calculations, especially when working with norms and distances.
* ** Orthogonality:* * Preserves all the advantages of an orthogonal base.

**Types of bases**
* **Cartesian base:** The most common type of base, where the axes are perpendicular to each other and have the same unit.
* **Spherical base:** Used to describe the position of points in spherical coordinates (radius, angle in the horizontal plane, angle from the vertical).
* **Cylindrical base:** Used to describe the position of points in cylindrical coordinates (radius, angle, height).



## Cartesian coordinate system

Since we perceive the surrounding space as three-dimensional, a suitable coordinate system is the [Cartesian coordinate system]( http://en.wikipedia.org/wiki/Cartesian_coordinate_system) in the [Cartesian coordinate system]( http://en.wikipedia.org/wiki/Cartesian_coordinate_system) with three orthogonal axes, as shown below. The orthogonality of the Cartesian coordinate system is convenient for its use in classical mechanics, it is usually assumed that the structure of space has a [Cartesian coordinate system]( http://en.wikipedia.org/wiki/Cartesian_coordinate_system) and, as a result, motions in different directions are independent of each other.

<div class='center-align'><figure><img src="https://raw.githubusercontent.com/demotu/BMC/master/images/CCS.png" width=350/><figcaption><center><i>Figure. A point in three-dimensional Euclidean space described in a Cartesian coordinate system.< /i></center></figcaption> </figure></div>

Another initiative for standardizing reference frames comes from the [Virtual Animation of the Kinematics of the Human for Industrial, Educational and Research Purposes (VAKHUM )] project ( https://github.com/BMClab/BMC/blob/master/courses/refs/VAKHUM.pdf ).

## Determining the coordinate system

The most common coordinate system we use is the Cartesian coordinate system, which uses three mutually perpendicular axes (x, y, z) and an origin. Every point in space can be uniquely identified by three numbers that indicate the distances from these axes.

<div class='center-align'><figure><img src="https://raw.githubusercontent.com/demotu/BMC/master/images/vector3Dijk.png" width=350/><figcaption><center><i>Figure. Representation of point <b>P</b> and its position vector $\overrightarrow{\mathbf{r }}$ in a Cartesian coordinate system. The bases $\vec{\mathbf{i}}, \vec{\mathbf{j}}, \vec{\mathbf{k }}$ form the base of this coordinate system and are usually represented in the RGB (red, green, blue) color sequence for easier visualization.</i></center></figcaption></figure></div>

It can be seen that the base versors shown in the image above have the following coordinates in the Cartesian coordinate system:
<br>
<span class="notranslate">
\begin{equation}
\hat{\mathbf{i}} = \begin{bmatrix}1\\0\\0 \end{bmatrix}, \quad \hat{\mathbf{j}} = \begin{bmatrix}0\\1\\0 \end{bmatrix}, \quad \hat{\mathbf{k}} = \begin{bmatrix} 0 \\ 0 \\ 1 \end{bmatrix}
\end{equation}
</span>

Using the notation described in the figure above, the position vector $\overrightarrow{\mathbf{r }}$ (or point $\overrightarrow{\mathbf{P}}$) can be expressed as:
<br>
<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{r}} = x \vec{\mathbf{i}} + y \vec{\mathbf{j}} + z\vec{\mathbf{k}}
\end{equation}
</span>

### Definition of base

The mathematical problem of determining a coordinate system is to find a base and an origin for it (a base is just a set of vectors without an origin). There are various methods for calculating a base given by a set of points (coordinates), for example, the scalar product or the vector product can be used for this problem.

#### Using the cross product

Let's now define the base using the common method in motion analysis (using the cross product):
Given the coordinates of three non-collinear points in 3D space (points that do not all lie on the same line), $\overrightarrow{\mathbf{m }}_ 1, \overrightarrow{\mathbf{m}}_2, \overrightarrow{\mathbf{m}}_3$, which can represent, for example, the position of markers captured from a motion analysis session, the following steps can be used:

1. The first axis, $\overrightarrow{\mathbf{v }}_ 1$, the vector $\overrightarrow{\mathbf{m}}_2-\overrightarrow{\mathbf{m}}_1$ (or any other vector difference);

2. The second axis, $\overrightarrow{\mathbf{v }}_ 2$, the vector product between the vectors $\overrightarrow{\mathbf{v}}_1$ and $\overrightarrow{\mathbf{m}}_3-\overrightarrow{\mathbf{m}}_1$ (or $\overrightarrow{\mathbf{m}}_3-\overrightarrow{\mathbf{m}}_2$);

3. The third axis, $\overrightarrow{\mathbf{v }}_ 3$, is the cross product between the vectors $\overrightarrow{\mathbf{v}}_1$ and $\overrightarrow{\mathbf{v}}_2$; and

4. Normalizing vectors so that they have norm 1, dividing each vector by its norm.

The positions of the points used to construct the coordinate system must, by definition, be specified in relation to an existing coordinate system. In motion analysis, this coordinate system is the coordinate system from the motion capture system and is determined in the calibration phase. In this phase, the positions of markers placed on the object with perpendicular axes and known distances between markers are captured and used as a reference (laboratory) coordinate system.

For example, given the positions $\overrightarrow{\mathbf{m }}_ 1 = [1,2,5], \overrightarrow{\mathbf{m}}_2 = [2,3,3], \overrightarrow{\mathbf{m}}_3 = [4,0,2]$, a base can be found with:

In [3]:
import numpy as np

m1 = np.array ([1, 2, 5])
m2 = np.array ([2, 3, 3])
m3 = np.array ([4, 0, 2])

v1 = m2 - m1 # first axis
v2 = np.cross(v1, m3 - m1) # second axis
v3 = np.cross(v1, v2) # third axis

# Vector normalization
e1 = v1/np.linalg.norm(v1)
e2 = v2/np.linalg.norm(v2)
e3 = v3/np.linalg.norm(v3)

print( 'Versors:', '\ne1 =', e1, '\ne2 =', e2, '\ne3 =', e3)

print( '\nNorm of each of the versors:',
'\n||e1|| =', np.linalg .norm(e1),
'\n||e2|| =', np.linalg .norm(e2),
'\n||e3|| =', np.linalg .norm(e3))

print( '\nTest of orthogonality (using cross product):',
'\ne1 x e2:', np.linalg .norm(np.cross(e1, e2)),
'\ne1 x e3:', np.linalg .norm(np.cross(e1, e3)),
'\ne2 x e3:', np.linalg .norm(np.cross(e2, e3)))

Versors: 
e1 = [ 0.40824829  0.40824829 -0.81649658] 
e2 = [-0.76834982 -0.32929278 -0.5488213 ] 
e3 = [-0.49292179  0.85141036  0.17924429]

Norm of each of the versors: 
||e1|| = 1.0 
||e2|| = 1.0 
||e3|| = 1.0

Test of orthogonality (using cross product): 
e1 x e2: 1.0 
e1 x e3: 1.0000000000000002 
e2 x e3: 0.9999999999999999


Due to rounding errors ([more here]( https://en.wikipedia.org/wiki/Round-off_error)) and inaccuracies in calculations with floating-point decimal numbers ([more here](https://en.wikipedia.org/wiki/Round-off_error)), the norms do not exactly equal 1.
If necessary, we can round the result:

In [4]:
print( '\nTest of orthogonality (using cross product):',
'\ne1 x e2:', np.linalg .norm(np.cross(e1, e2)).round(8),
'\ne1 x e3:', np.linalg .norm(np.cross(e1, e3)).round(8),
'\ne2 x e3:', np.linalg .norm(np.cross(e2, e3)).round(8))


Test of orthogonality (using cross product): 
e1 x e2: 1.0 
e1 x e3: 1.0 
e2 x e3: 1.0


Or format the text representing the result:

In [5]:
print( '\nTest of orthogonality (using cross product):',
f'\ne1 x e2: { np.linalg .norm(np.cross(e1, e2)):g}',
f'\ne1 x e3: { np.linalg .norm(np.cross(e1, e3)):g}',
f'\ne2 x e3: { np.linalg .norm(np.cross(e2, e3)):g}')


Test of orthogonality (using cross product): 
e1 x e2: 1 
e1 x e3: 1 
e2 x e3: 1


Where we used `f'{ :.g }'` to format a number to a string ([more details here](https://docs.python.org/dev/library/stdtypes.html#printf-style-string-formatting) [or here](https://docs.python.org/dev/library/stdtypes.html#printf-style-string-formatting)).

For a permanent change, we could use the numpy settings to represent the decimal point precision with the function [`np.set_printoptions `]( https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html), but that only works for numpy arrays, not numbers. Or use the IPython magic command [`np.set_printoptions `]( https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html), but in this case you cannot use the print function to display the result.

### Definition of the origin

In order to define a coordinate system using a calculated base, we also need to define an origin. In principle, we could use any point as the origin, but if the calculated coordinate system should follow anatomical conventions, e.g. the origin of the coordinate system should be at the center of a joint, we will need to calculate the base and origin according to the standards used in motion analysis.

When the coordinate system is a technical base rather than an anatomical one, a common practice in motion analysis is to define the origin of the coordinate system as the centroid (average) position between the markers in the reference frame. Using the average position across the markers potentially reduces the influence of noise (such as from soft tissue artifact) on the calculation.

For the marks in the example above, the origin of the coordinate system will be:

In [6]:
origin = np.mean ((m1, m2, m3), axis=0)
print( 'Origin: ', origin)

Origin:  [2.33333333 1.66666667 3.33333333]


Draw the coordinate system and base using Python's custom function `CCS.py`.
We can copy and paste the contents of the `CCS.py` file here (and run the cell) or we can load the function from a directory, let's do that. First, create a "functions" folder and place the `CCS.py` file in it.

In [ ]:
path2 = r'./functions'
import sys
sys.path.insert(1, path2)  # přidat do pythonpath
from CCS import CCS

In [7]:
markers = np.vstack ((m1, m2, m3))
base = np.vstack ((e1, e2, e3))

In [13]:
!pip3 install -q ipympl

### Gram–Schmidt process

Another classic mathematical procedure using the scalar product is known as the [Gram–Schmidt process]( http://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process). See the [Gram–Schmidt process]( http://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process) wikipedia for an example of the Gram–Schmidt process and how to implement it in Python.

The [Gram–Schmidt process]( http://en.wikipedia.org/wiki/Gram%E2%80%93Schmidt_process) is a method for orthonormalizing (orthogonal unit versors) a set of vectors using a dot product. The Gram–Schmidt process works for any number of vectors.

For example, given three vectors <span class="notranslate">
$\overrightarrow{\mathbf{a}}, \overrightarrow{\mathbf{b}}, \overrightarrow{\mathbf{c }}$ </span>, in 3D space the base $\{\vec{e}_a, \vec{e}_b, \vec{e}_c\}$ can be found using the Gram–Schmidt process

The first versor is in <span class="notranslate">
$\overrightarrow{\mathbf{a }}$ </span> direction (or in the direction of any of the other vectors):
<br>
<span class="notranslate">
\begin{equation}
\vec{e}_a = \frac{\overrightarrow{\mathbf{a} }}{ ||\overrightarrow{\mathbf{a}}||}
\end{equation}
</span>

The second versor, orthogonal to $\vec{e}_a$, can be found assuming that we can express the vector $\overrightarrow{\mathbf{b }}$ in the direction of $\vec{e}_a$ as:
<br>
<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{b}} = \overrightarrow{\mathbf{b }}^ \| + \overrightarrow{\mathbf{b}}^\bot
\end{equation}
</span>

Then:

<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{b }}^ \bot = \overrightarrow{\mathbf{b}} - \overrightarrow{\mathbf{b}}^\| = \overrightarrow{\mathbf{b}} - (\overrightarrow{\mathbf{b}} \cdot \vec{e}_a ) \vec{e}_a
\end{equation}
</span>

Finally:

<span class="notranslate">
\begin{equation}
\vec{e}_b = \frac{\overrightarrow{\mathbf{b }}^ \bot}{||\overrightarrow{\mathbf{b}}^\bot||}
\end{equation}
</span>

The third versor, orthogonal to $\{\vec{e}_a, \vec{e}_b \}$ , can be found expressing the vector $\overrightarrow{\mathbf{C}}$ in the directions $\vec{e}_a$ and $\vec{e}_b$ as:
<br>
<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{c}} = \overrightarrow{\mathbf{c }}^ \| + \overrightarrow{\mathbf{c}}^\bot
\end{equation}
</span>

Then:
<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{c }}^ \bot = \overrightarrow{\mathbf{c}} - \overrightarrow{\mathbf{c}}^\|
\end{equation}
</span>

Where:
<span class="notranslate">
\begin{equation}
\overrightarrow{\mathbf{c }}^ \| = (\overrightarrow{\mathbf{c}} \cdot \vec{e}_a ) \hat{e}_a + (\overrightarrow{\mathbf{c}} \cdot \vec{e}_b ) \hat{e}_b
\end{equation}
</span>

Finally:
<span class="notranslate">
\begin{equation}
\vec{e}_c = \frac{\overrightarrow{\mathbf{c }}^ \bot}{||\overrightarrow{\mathbf{c}}^\bot||}
\end{equation}
</span>

And an animation of the Gram-Schmidt process in progress:

<div class='center-align'><figure><img src="https://upload.wikimedia.org/wikipedia/commons/e/ee/Gram-Schmidt_orthonormalization_process.gif" width=450/><figcaption><center><i>Figure. The Gram-Schmidt process is performed on three base vectors for R3 (<a href="https://commons.wikimedia.org/w/index.php?curid=24396471">image from Wikipedia</a> ).</i></center></figcaption></figure></div>

Let's implement the Gram–Schmidt process in Python.

For example, given the positions we saw earlier, $\overrightarrow{\mathbf{m }}_ 1 = [1,2,5], \overrightarrow{\mathbf{m}}_2 = [2,3,3], \overrightarrow{\mathbf{m}}_3 = [4,0,2]$, the base can be found with:

The first version is:

In [19]:
ea = m1/np.linalg.norm(m1)
print(ea)

[0.18257419 0.36514837 0.91287093]


The second versor is:

In [20]:
eb = m2 - np.dot( m2, ea)*ea
eb = eb/np.linalg.norm(eb)
print(eb)

[ 0.59020849  0.70186955 -0.39878952]


And the third version is:

In [21]:
ec = m3 - np.dot( m3, ea)*ea - np.dot(m3, eb)*eb
ec = ec/np.linalg.norm(ec)
print(ec)

[ 0.78633365 -0.61159284  0.08737041]


Let's check the orthonormality between these versors:

In [22]:
print( " Versors:", "\nea =", ea, "\neb =", eb, "\nec =", ec)

print(
"\n Standard of each version:",
"\n ||ea|| =",
    np.linalg.norm (ea),
"\n ||eb|| =",
    np.linalg.norm (eb),
"\n ||ec|| =",
    np.linalg .norm(ec),
)

print(
"\n Orthogonality test (scalar product):",
"\n ea . eb:",
    np.dot( ea, eb),
"\n eb . ec:",
    np.dot( eb, ec),
"\n ec . ea:",
    np.dot( ec, ea),
)

 Versors: 
ea = [0.18257419 0.36514837 0.91287093] 
eb = [ 0.59020849  0.70186955 -0.39878952] 
ec = [ 0.78633365 -0.61159284  0.08737041]

 Standard of each version: 
 ||ea|| = 1.0 
 ||eb|| = 1.0 
 ||ec|| = 1.0

 Orthogonality test (scalar product): 
 ea . eb: -3.3306690738754696e-16 
 eb . ec: 6.245004513516506e-17 
 ec . ea: 8.326672684688674e-17


## 

Spherical coordinate system

**Spherical coordinates** are a three-dimensional coordinate system that uses to determine the position of a point in space:

* **Radial distance ($\rho$ ):** The distance of a point from the origin.
* **Azimuthal angle ($\theta$ ):** The angle in the xy plane between the positive x-axis and the projection of a point onto this plane.
* **Zenith angle ($\phi$ ):** The angle between the positive axis behind the line connecting the point with the origin.

![ Spherical Coordinate System](https://cdn1.byjus.com/wp-content/uploads/2019/09/Spherical-coordinate-system.png)

**Spherical Coordinate Base**

The base of spherical coordinates consists of three unit vectors:

* **$\vec{\mathbf{e }}_ \rho$:** Directs radially from the origin to the point.
* **$\vec{\mathbf{e }}_ \theta$:** Points in the direction of increasing azimuthal angle $\theta$.
* **$\vec{\mathbf{e }}_ \varphi$:** Points in the direction of increasing zenith angle $\varphi$.

This base is orthonormal, meaning that the vectors are mutually perpendicular and have unit magnitude.

**Transformation from Cartesian coordinates**

To convert Cartesian coordinates ($x, y, z$) to spherical coordinates ($\rho, \theta, \varphi$), the following relations apply:

* $\rho = \ sqrt{ x^2 + y^2 + z^2}$
* $\theta = \arctan{\frac{y}{x }}$
* $\varphi = \arccos{\frac{z}{\rho }}$

**Transformation to Cartesian coordinates**

The following relations apply to the conversion of spherical coordinates to Cartesian coordinates:
* $x = \rho \sin{\varphi} \cos{\ theta}$
* $y = \rho \sin{\varphi} \sin{\ theta}$
* $z = \rho \cos{\ varphi}$

**Advantages of spherical coordinates:**

* ** Symmetry:** They are used advantageously for problems with spherical symmetry (e.g. gravitational field, electrostatic field).
* **Simplification of calculations:** In some cases, they can simplify the solution of physics problems.

**Disadvantages of spherical coordinates: **

* **More complex derivatives:** Calculating gradient, divergence, and rotation in spherical coordinates is more complex than in Cartesian coordinates.
* ** Singularities:* * Singularities can occur at the origin (ρ = 0) and at the poles (φ = 0, π).

IFrame( 'https://www.geogebra.org/classic/c5nsjx3t?embed', width='100%', height=500)

## Cylindrical coordinates

Cylindrical coordinates are suitable for systems with cylindrical symmetry. Three coordinates are used:

* **Radial distance (ρ ):** The distance of a point from the z-axis.
* **Azimuthal angle (θ ):** The angle in the xy plane measured from the positive x-axis.
* **Height (z ):** The distance of a point from the xy plane (same as in Cartesian coordinates).

![ Cylindrical coordinates](https://upload.wikimedia.org/wikipedia/commons/0/0e/Coord_system_CY_1.svg)

**Transformational relationships: **

* **From cylindrical to Cartesian:**
* $x = \rho \cos{\ theta}$
* $y = \rho \sin{\ theta}$
* $z = z$
* **From Cartesian to Cylindrical:**
* $\rho = \ sqrt{ x^2 + y^2}$
* $\theta = \arctan{\frac{y}{x }}$
* $z = z$

In [6]:
IFrame( 'https://www.geogebra.org/classic/X3j28ZkC?embed', width='100%', height=500)

** Comparison: **

| Property | Cartesian | Spherical | Cylindrical |
|---|---|---|---|
| Symmetry | Rectangular | Spherical | Cylindrical |
| Coordinates | x, y, z | ρ, θ, φ | ρ, θ, z |
| Usage | General | Spherical symmetry | Cylindrical symmetry |

**When to use which system?**

* **Cartesian:** Most everyday problems, linear motion.
* **Spherical:** Problems with spherical symmetry (e.g. gravity, electrostatics).
* **Cylindrical:** Problems with cylindrical symmetry (e.g. fluid flow in pipes).

**Why use coordinate systems other than Cartesian?**

* **Simplification of calculations:** Problems with certain symmetry may be simpler in other coordinate systems.
* **More natural description:** Some physical phenomena are more naturally described in other coordinate systems.


## Further reading

- [The right frame of reference makes it simple: An example of introductory mechanics supported by video analysis of motion]( https://www.researchgate.net/publication/267761615_The_right_frame_of_reference_makes_it_simple_An_example_of_introductory_mechanics_supported_by_video_analysis_of_motion)


## Video lectures on the Internet

- [Introduction to reference - Khan Academy]( https://www.khanacademy.org/science/physics/one-dimensional-motion/displacement-velocity-time/v/introduction-to-reference-frames)
- [Introduction to orthonormal bases - Khan Academy]( https://www.khanacademy.org/math/linear-algebra/alternate-bases/orthonormal-base/v/linear-algebra-introduction-to-orthonormal-bases)
- [The Gram-Schmidt process - Khan Academy]( https://www.khanacademy.org/math/linear-algebra/alternate-bases/orthonormal-base/v/linear-algebra-the-gram-schmidt-process)
- [Biomechanics of Movement | Demo: Motion Capture Placement and Reference Frames]( https://youtu.be/ctwoH59Obew)

## Problems

1. How fast are you moving right now? In your answer, consider your motion relative to the Earth and relative to the Sun.

2. Go to the website [ http://www.wisc-online.com/Objects/ViewObject.aspx?ID=AP15305]( http://www.wisc-online.com/Objects/ViewObject.aspx?ID=AP15305 ) and complete the interactive lesson to learn about anatomical terminology for describing relative position in the human body.

3. To learn more about Cartesian coordinate systems, go to the website [ http://www.mathsisfun.com/data/cartesian-coordinates.html]( http://www.mathsisfun.com/data/cartesian-coordinates.html ), study the material, and answer the 10 questions at the end.

4. Given points in 3D space, m1 = [2, 2, 0], m2 = [0, 1, 1], m3 = [1, 2, 0], find the orthonormal base.

5. Determine whether the following points form a base in 3D space, m1 = [2, 2, 0], m2 = [1, 1, 1], m3 = [1, 1, 0].

6. Derive expressions for the three axes of the pelvis, taking into account the convention of the project [Virtual Animation of the Kinematics of the Human for Industrial, Educational and Research Purposes (VAKHUM )]( https://github.com/BMClab/BMC/blob/master/courses/refs/VAKHUM.pdf) (each use RASIS, LASIS, RPSIS and LPSIS as names for the pelvic axis landmark and expression).

7. Determine the vector base for the pelvis according to the convention of the project [Virtual Animation of the Kinematics of the Human for Industrial, Educational and Research Purposes (VAKHUM )]( https://github.com/BMClab/BMC/blob/master/courses/refs/VAKHUM.pdf) for the following anatomical landmark positions (units in meters): RASIS=[0.5, 4] 0.8, 70 . 0.1], RPSIS = [0.3, 0.85, 0.2], LPSIS = [0.29, 0.78, 0.3].

## References

- Corke P (2017) [Robotics, Vision and Control: Fundamental Algorithms in MATLAB]( https://petercorke.com/RVC/). 2nd ed. Springer-Verlag Berlin.
- [Standards - International Society of Biomechanics]( https://isbweb.org/activities/standards).
- The Stanford Encyclopedia of Philosophy. [Space and Time: Inertial Frames]( http://plato.stanford.edu/entries/spacetime-iframes/).
- [Virtual Animation of the Kinematics of the Human for Industrial, Educational and Research Purposes (VAKHUM )]( https://github.com/BMClab/BMC/blob/master/courses/refs/VAKHUM.pdf).